### PR TITLE
[Refactor:TAGrading] Hide custom marks

### PR DESCRIPTION
**Pull request by KKhaghani**
### Please check if the PR fulfills these requirements:

* [ ] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
Gradeables, peer and non-peer, can be given custom marks complete with custom messages and a custom point value. 
Peers can also click "edit rubric" on a peer gradeable and make changes.

### What is the new behavior?
Works on #5328
Peer gradeables no longer display the custom mark option.
All additional code concerning _non_-peer gradeables, which acted as groundwork to continue #5328, has been removed.
Peers can no longer edit the rubric or add custom marks when they are grading but if a TA or Instructor is grading a peer assignment, they can still do both those things.

### CODE REVIEW

**Review by vikrsri — APPROVED** [review]
Changes work as proposed, students can no longer add custom marks on peer gradeables.

File: site/public/js/ta-grading-rubric.js [hunk 1]
```diff
@@ -2929,7 +2929,7 @@ function saveComponent(component_id) {
         // The grader unchecked the custom mark, but didn't delete the text.  This shouldn't happen too often,
         //  so prompt the grader if this is what they really want since it will delete the text / score.
         let gradedComponent = getGradedComponentFromDOM(component_id);
-        if (gradedComponent.comment !== '' && !gradedComponent.custom_mark_selected) {
+        if (gradedComponent.custom_mark_visible && gradedComponent.comment !== '' && !gradedComponent.custom_mark_selected) {
```
Comment (MasterOdin) [inline site/public/js/ta-grading-rubric.js:2932]: Is `custom_mark_visible` defined anywhere? It does not look like it?

**Review by MasterOdin — CHANGES_REQUESTED** [review]

File: site/public/js/ta-grading-rubric.js [hunk 2]
```diff
@@ -2929,7 +2929,7 @@ function saveComponent(component_id) {
         // The grader unchecked the custom mark, but didn't delete the text.  This shouldn't happen too often,
         //  so prompt the grader if this is what they really want since it will delete the text / score.
         let gradedComponent = getGradedComponentFromDOM(component_id);
-        if (gradedComponent.comment !== '' && !gradedComponent.custom_mark_selected) {
+        if (gradedComponent.custom_mark_enabled && gradedComponent.comment !== '' && !gradedComponent.custom_mark_selected) {
```
Comment (MasterOdin) [inline site/public/js/ta-grading-rubric.js:2932]: This variable is also undefined though?

The return from `getGradedComponentFromDOM` is:

https://github.com/Submitty/Submitty/blob/8e7563346c2522014aeb9c38bb689a36dc3dfa46/site/public/js/ta-grading-rubric.js#L1262-L1271

I'm guessing you'll want to modify that function to do an existence check on the custom mark field and return that as a boolean as `custom_mark_enabled` within that function.

Comment (KKhaghani) [reply site/public/js/ta-grading-rubric.js:2932]: After defining the variable in `getGradedComponentFromDOM` I noticed a bug that had no doubt been present in my custom-mark fix from the start:
On a gradeable with both peer and non-peer components, opening a non-peer component then opening a peer component allows the custom mark to be selected (and is visible) on peer components. A refresh fixes this.
Although I am dealing with some high-priority assignments at the moment, I hope to commit a fix for this bug in the next few days, as well as the variable definition.

Comment (loadedskiffer) [reply site/public/js/ta-grading-rubric.js:2943]: @MasterOdin I believe I have fixed these problems, could you take another look ?

**Review by A3Digital — CHANGES_REQUESTED** [review]
It does look like students can no longer edit rubrics. However, it seems like my students still have the custom mark option available.

Custom Mark

**Comment by bmcutler** [comment]
closing this -- replaced by #6551

**Pull request closed without merging**